src/common/arrow/arrow_row_batch.cpp

```diff
@@ -15,22 +16,22 @@ ArrowRowBatch::ArrowRowBatch(
     }
 }
 
-template<DataTypeID DT>
+template<LogicalTypeID DT>
```

Comment:
I think we can use PhysicalType here, and it would be better than LogicalTypeID. You can add a todo comment here or open an issue. We will clean this up later.